app/controllers/comments_controller.rb

```diff
@@ -274,6 +269,26 @@ def admin_delete
 
   private
 
+  def comment_should_be_visible?
+    if @article
+      @article.published?
+    else
+      @root_comment
+    end
+  end
```

Comment:
Should we check for `@commentable` too? Or would `@article` and `@root_comment` be enough?

Reply:
there's actually an expected case for @commentable is nil (deleted article) but we want to show the comment by comment.path.

That's kind of like `@root_comment && @root_comment.commentable.nil?` being true, but it's handled in the view by conditionals when root comment but not article - the heading turns into "Comments for a deleted post" (in the test case). The controller doesn't need to focus on that possibility since the view handles it already.